### DISCUSSION TITLE
如何编写正确的 test case

### DISCUSSION THREAD

**Discussion by yangxiuxiu1115** [ant-design/ant-design#42653] [Q&A]
我正在尝试解决这个 [bug](https://github.com/ant-design/ant-design/issues/42551)，但是我在补充test case过程中遇见了一些问题，或许是我对 `jest` 和 `enzyme` 不太熟悉的原因，有人能够帮助我吗。我尝试写的 case 如下所示：
 ``` 
it('should be opened correctly when using dynamic maxCountTag and onDropdownVisibleChange', async () => {
    const onDropdownVisibleChange = jest.fn();
    const Test = () => {
      const [isOpen, setIsOpen] = React.useState(false);
      return (
        <Select
          mode="multiple"
          open={isOpen}
          onDropdownVisibleChange={(o) => {
            setIsOpen(o);
            onDropdownVisibleChange();
          }}
          value={['1', '2']}
          maxTagCount={isOpen ? 0 : 'responsive'}
        >
          <Option value="1">1</Option>
          <Option value="2">2</Option>
          <Option value="3">3</Option>
          <Option value="4">4</Option>
        </Select>
      );
    };
    const wrapper = mount(<Test />);
    wrapper.find('.rc-select-selection-overflow').simulate('mousedown');
    expectOpen(wrapper);
    expect(onDropdownVisibleChange).toBeCalledTimes(1);
  });
```

**Comment by BoyYangzai** [comment]
@RicardoErii 
看了下你的测试用例 我大概清楚你遇到了什么样的问题
试试给 Select 添加上 open prop，这样你就可以获取到你想要的 dom 节点了😄
props 文档在这：
https://ant.design/components/select-cn#select-props
感谢你给 antd 做出贡献👍

**Reply by yangxiuxiu1115** [reply]
Hi, @BoyYangzai ，十分感谢你的回复。
这是一个 `rc-select` 的测试用例。我期望能够根据`select`的`open`状态去切换`maxTagCount`的值，因此我创建了一个`isOpen`的`state`。
在上述的测试用例中，我能够获取想要的`dom`节点，但是在触发后续的模拟事件中与实际页面并不一致，可以跳转到我上面提供的 [issue](https://github.com/ant-design/ant-design/issues/42551)。

**Reply by BoyYangzai** [reply]
@RicardoErii 设置 `open` 初始值为 `true 试一下，保证他刚开始dom渲染出来

**Reply by RedJue** [reply]
> Hi, @BoyYangzai ，十分感谢你的回复。
> 
> 这是一个 `rc-select` 的测试用例。我期望能够根据`select`的`open`状态去切换`maxTagCount`的值，因此我创建了一个`isOpen`的`state`。
> 
> 在上述的测试用例中，我能够获取想要的`dom`节点，但是在触发后续的模拟事件中与实际页面并不一致，可以跳转到我上面提供的 [issue](https://github.com/ant-design/ant-design/issues/42551)。
> 
> 

没找到节点可能和渲染时机有关系，试试通过 promise await 几百毫秒试试？

**Reply by yangxiuxiu1115** [reply]
我能找到节点，并且触发事件。但是在测试环境中`rc-select`与浏览器表现并不一致，好像在测试环境中全局所绑定的`mousedown`事件没有被冒泡触发。
